scripts/kind-with-registry.sh

```diff
@@ -42,37 +56,140 @@ if [ "$(docker inspect -f '{{.State.Running}}' "${reg_name}" 2>/dev/null || true
+  echo "Uploading openid-configuration document to '${AZWI_STORAGE_ACCOUNT}' storage account"
+  upload_to_blob "${AZWI_STORAGE_CONTAINER}" "${AZWI_OPENID_CONFIG_FILEPATH}" ".well-known/openid-configuration"
+  echo "Uploading jwks document to '${AZWI_STORAGE_ACCOUNT}' storage account"
+  upload_to_blob "${AZWI_STORAGE_CONTAINER}" "${AZWI_JWKS_JSON_FILEPATH}" "openid/v1/jwks"
+  echo "Removing key access on storage account as no further data writes are required"
+  az storage account update -n "${AZWI_STORAGE_ACCOUNT}" -g "${AZWI_RESOURCE_GROUP}" --subscription "${AZURE_SUBSCRIPTION_ID}" --allow-shared-key-access=false --output none --only-show-errors
+  fi
+  if [ -z "${AZURE_CLIENT_ID_USER_ASSIGNED_IDENTITY}" ]; then
+    if [ -z "${USER_IDENTITY}" ]; then
+        echo "USER_IDENTITY environment variable required if not bringing your own identity via AZURE_CLIENT_ID_USER_ASSIGNED_IDENTITY"
+        exit 1
+    fi
+    az identity create -n "${USER_IDENTITY}" -g "${AZWI_RESOURCE_GROUP}" -l "${AZWI_LOCATION}" --output none --only-show-errors --tags creationTimestamp="${TIMESTAMP}" jobName="${JOB_NAME}" buildProvenance="${BUILD_PROVENANCE}"
+    AZURE_IDENTITY_ID=$(az identity show -n "${USER_IDENTITY}" -g "${AZWI_RESOURCE_GROUP}" --query clientId -o tsv)
+    AZURE_IDENTITY_ID_PRINCIPAL_ID=$(az identity show -n "${USER_IDENTITY}" -g "${AZWI_RESOURCE_GROUP}" --query principalId -o tsv)
+    echo "${AZURE_IDENTITY_ID}" > "${AZURE_IDENTITY_ID_FILEPATH}"
+    until az role assignment create --assignee-object-id "${AZURE_IDENTITY_ID_PRINCIPAL_ID}" --role "Contributor" --scope "/subscriptions/${AZURE_SUBSCRIPTION_ID}" --assignee-principal-type ServicePrincipal --output none --only-show-errors; do
+      sleep 5
+    done
+    az identity federated-credential create -n "capz-federated-identity" \
+      --identity-name "${USER_IDENTITY}" \
+      -g "${AZWI_RESOURCE_GROUP}" \
+      --issuer "${SERVICE_ACCOUNT_ISSUER}" \
+      --subject "system:serviceaccount:capz-system:capz-manager" --output none --only-show-errors
+    az identity federated-credential create -n "aso-federated-identity" \
+      --identity-name "${USER_IDENTITY}" \
+      -g "${AZWI_RESOURCE_GROUP}" \
+      --issuer "${SERVICE_ACCOUNT_ISSUER}" \
+      --subject "system:serviceaccount:capz-system:azureserviceoperator-default" --output none --only-show-errors
   fi
-  mkdir -p "$HOME"/azwi/creds
-  echo "${SERVICE_ACCOUNT_SIGNING_PUB}" > "$HOME"/azwi/creds/sa.pub
-  echo  "${SERVICE_ACCOUNT_SIGNING_KEY}" > "$HOME"/azwi/creds/sa.key
-  SERVICE_ACCOUNT_ISSUER="${SERVICE_ACCOUNT_ISSUER:-https://oidcissuercapzci.blob.core.windows.net/oidc-capzci/}"
+}
+
+function upload_to_blob() {
+  local container_name=$1
+  local file_path=$2
+  local blob_name=$3
+
+  echo "Uploading ${file_path} to '${AZWI_STORAGE_ACCOUNT}' storage account"
+  az storage blob upload \
+      --container-name "${container_name}" \
```

Comment:
you could re-use "${AZWI_STORAGE_CONTAINER}" like `AZWI_STORAGE_ACCOUNT`, since its going to same place in both instances

Reply:
good call